subprojects/snapshots/src/test/groovy/org/gradle/internal/snapshot/AbstractFileSystemNodeWithChildrenTest.groovy

```diff
@@ -60,10 +60,14 @@ abstract class AbstractFileSystemNodeWithChildrenTest<NODE extends FileSystemNod
         this.selectedChildPath = spec.selectedChildPath
         if (selectedChildPath != null) {
             def selectedChildIndex = indexOfSelectedChild
-            this.selectedChild = selectedChildIndex == -1 ? null : children.entries().get(selectedChildIndex).value
+            this.selectedChild = selectedChildIndex == -1 ? null : entriesFor(children).get(selectedChildIndex).value
```

Comment:
We are collecting to a list here just to find the `selectedChildIndex`th element? That sounds like something we could do much cheaper.

Reply:
Ah, this is only a test...

Reply:
Seems like we could use a `getNthChild()` method instead in other places, too.